### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Valdemar3X**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/model/User.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    private String password;
+
+    @Column(nullable = false)
+    private String firstName;
+
+    @Column(nullable = false)
+    private String lastName;
+
+    private String shippingAddress;
+
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/model/User.java]:
```suggestion
```

File: src/main/java/mate/academy/bookstore/dto/user/UserRegistrationRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package mate.academy.bookstore.dto.user;
+
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import lombok.Data;
+import mate.academy.bookstore.validation.FieldMatch;
+
+@FieldMatch(field = "password",
+        fieldMatch = "repeatPassword",
+        message = "Password must match")
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/dto/user/UserRegistrationRequestDto.java]:
```suggestion
        message = "Password fields must match")
```

File: src/main/java/mate/academy/bookstore/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            throws RegistrationException {
+        if (repository.findByEmail(request.getEmail()).isPresent()) {
+            throw new RegistrationException("Unable to complete registration.");
+        }
+        User user = new User();
+        user.setEmail(request.getEmail());
+        user.setPassword(request.getPassword());
+        user.setFirstName(request.getFirstName());
+        user.setLastName(request.getLastName());
+        user.setShippingAddress(request.getShippingAddress());
```
Comment (okuzan) [inline src/main/java/mate/academy/bookstore/service/impl/UserServiceImpl.java]: it's a job for a mapper

File: src/main/resources/db/changelog/changes/02-create-users-table.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                  constraints:
+                    nullable: false
+              - column:
+                  name: last_name
+                  type: varchar(255)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: shipping_address
+                  type: varchar(255)
```
Comment (okuzan) [inline src/main/resources/db/changelog/changes/02-create-users-table.yaml]: image
always review your PR and add one empty line at EOF where you see this

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 5]
```diff
@@ -1,3 +1,5 @@
 databaseChangeLog:
   - include:
       file: db/changelog/changes/01-create-books-table.yaml
+  - include:
+      file: db/changelog/changes/02-create-users-table.yaml
```
Comment (okuzan) [inline src/main/resources/db/changelog/db.changelog-master.yaml]:
```suggestion
      file: db/changelog/changes/02-create-users-table.yaml
      
```

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by kozhukhovsky — APPROVED** [review]

**Pull request merged**